en/concept/architecture.md

```diff
@@ -12,9 +12,24 @@ All PX4 [airframes](../airframes/README.md) share a single codebase (this includ
 In addition to these runtime considerations, its modularity maximizes [reusability](https://en.wikipedia.org/wiki/Reusability).
 
 
-## High Level Software Architecture
+## High-Level Software Architecture
 
-The diagram below provides an overview of the building blocks of PX4. 
+The following diagram shows a high-level overview for the building blocks of
+the flight stack. It contains the full pipeline from sensors, RC input and
+autonomous flight control (Navigator), down to the motor or servo control
+(Actuators).
+
+![PX4 High-Level Architecture](../../assets/diagrams/PX4_High-Level_Architecture.svg)
```

Comment:
Does this render OK in gitbook? ON github it has some oddities: 
![image](https://user-images.githubusercontent.com/5368500/33510413-d72f75ae-d75f-11e7-98b8-2291cc4aa08e.png)

Reply:
Yes. Interestingly it renders fine on github too with the direct link: https://github.com/PX4/Devguide/blob/3ab7ead214dded46ddeaa3f02cbcda27fce3cbc4/assets/diagrams/PX4_High-Level_Architecture.svg